### PR TITLE
Add metadata for core mode packages

### LINKED ISSUE #6317: Core packages should specify whether to be included in main build
**Describe the bug**
Currently all packages that aren't private or named `metapackage` are included in the main bundle.  We need to have a separate metadata field for whether to include the package.   cf https://github.com/jupyterlab/jupyterlab/pull/6314#issuecomment-489787707.

### PR DESCRIPTION
**Pull request by blink1073**
## References
Fixes #6317.

## Code changes
Adds metadata to non-extension packages imported directly in `dev_mode/index.js`.
Changes the criteria for inclusion in `dev_mode/package.json`.

## User-facing changes
None

## Backwards-incompatible changes
None

### CODE REVIEW

File: packages/coreutils/package.json [hunk 1]
```diff
@@ -50,5 +50,8 @@
   },
   "publishConfig": {
     "access": "public"
+  },
+  "jupyterlab": {
+    "include": true
```
Comment (ian-r-rose) [inline packages/coreutils/package.json]: Is this intended?

**Comment by blink1073** [comment]
This metadata is only for core packages ending up in dev_mode and core mode.  Happy to swap the logic since there are fewer that should be excluded.

File: packages/coreutils/package.json (hunk 1 above)
Comment (blink1073) [reply packages/coreutils/package.json]: Nope. 😁

**Comment by vidartf** [comment]
Is there any development pattern were it would be useful to put `core=true` in any package outside of the official core? Either in temporary testing modes, or in a released package? Asking purely for future reference :)

**Comment by blink1073** [comment]
I'm not following the use pattern for third party modules. I think add:sibling and our prereleases cover both of those use cases.

**Comment by blink1073** [comment]
I updated the logic so that extensions are automatically included unless marked otherwise, and non-extensions need `coreDependency: true`.  That meant only two of our packages needed to be marked.

File: buildutils/src/ensure-repo.ts [hunk 2]
```diff
@@ -120,7 +120,23 @@ function ensureJupyterlab(): string[] {
...
+      return;
+    }
+    // Skip if explicitly marked as not a core dep.
+    if (
+      'coreDependency' in data.jupyterlab &&
+      !data.jupyterlab.coreDependency
+    ) {
+      return;
+    }
+    // Skip if it is not makred as an extension or a core dep.
```
Comment (ian-r-rose) [inline buildutils/src/ensure-repo.ts]: "makred"

**Comment by ian-r-rose** [comment]
I guess it is more correct to include `application` and `coreutils`, since `index.js` explicitly depends on them. Would it be fair to say that it would still work if they were implicitly included?

Also, should we include the `dev_mode/package.json` integrity updates in this PR, or is that something that is best left for later?

**Comment by blink1073** [comment]
Yes, I made them explicit because they are imported directly.  See Zen of Python 😀.  Sure, I'll do the integrity update and the spelling fix.

**Comment by ian-r-rose** [comment]
Okay, sounds good!

**Comment by blink1073** [comment]
Docs failure is a rate-limit from Github, merging.

**Pull request merged**